conda/cli/install.py

```diff
@@ -242,68 +245,61 @@ def install(args, parser, command='install'):
         print_activate(args.name if args.name else prefix)
         return
 
-    try:
-        update_modifier = context.update_modifier
-        if isinstall and args.revision:
-            index = get_index(channel_urls=index_args['channel_urls'],
-                              prepend=index_args['prepend'], platform=None,
-                              use_local=index_args['use_local'], use_cache=index_args['use_cache'],
-                              unknown=index_args['unknown'], prefix=prefix,
-                              repodata_fn=args.repodata_fn)
-            unlink_link_transaction = revert_actions(prefix, get_revision(args.revision), index)
-        else:
-            solver = Solver(prefix, context.channels, context.subdirs, specs_to_add=specs,
-                            repodata_fn=args.repodata_fn)
-            if isinstall and args.update_modifier == NULL:
-                # try to do a quick solve with then existing env frozen by default
-                update_modifier = UpdateModifier.FREEZE_INSTALLED
-            if isupdate:
-                deps_modifier = context.deps_modifier or DepsModifier.UPDATE_SPECS
+    repodata_fns = args.repodata_fns
+    if not repodata_fns:
+        repodata_fns = ["current_repodata.json", "repodata.json"]
+    elif REPODATA_FN not in repodata_fns:
+        repodata_fns.append(REPODATA_FN)
+
+    for repodata_fn in repodata_fns:
```

Comment:
Why does this check multiple repodatas to begin with opposed to using one and then falling back to regular repodata.json?

Reply:
It doesn't.  Each iteration of this loop runs the install process.  The loop exits after the first one succeeds.  None of them are checked prior to entering this loop.

https://github.com/conda/conda/pull/8741/files/84f93dfeda2143058ad93e3295b352de26372f15#diff-34bb8d01f991a5ffabae5e80b987c83fR283